docs-arrow/api-reference/vectors.md

```diff
@@ -2,36 +2,113 @@
 
 ## Overview
 
+
+## Usage
+
+Constructing new `Vector` instances is done through the static `from()` methods
+
+
+## Special Vectors
+
 ### Dictionary Arrays
 
-The Dictionary type is a special array type that is similar to a factor in R or a pandas.Categorical in Python. It enables one or more record batches in a file or stream to transmit integer indices referencing a shared dictionary containing the distinct values in the logical array. This is particularly often used with strings to save memory and improve performance.
+The Dictionary type is a special array type that enables one or more record batches in a file or stream to transmit integer indices referencing a shared dictionary containing the distinct values in the logical array. Later record batches reuse indices in earlier batches and add new ones as needed.
+
+A `Dictionary` is similar to a `factor` in R or a pandas, or "Categorical" in Python. It is is often used with strings to save memory and improve performance.
+
+
+### StructVector
```

Comment:
@trxcllnt I have tried to capture what we discussed and sift through the source for more details, but it would be awesome if you could do a quick pass on the more specialized vector types.